frontend/cypress/tests/codeView/codeView_filterGlob.js

```diff
@@ -0,0 +1,177 @@
+    cy.get('.radio-button--search')
+        .should('be.visible')
+        .click();
+
+    // filter by file type should be unchecked
+    cy.get('.radio-button--checkbox')
+        .should('not.be.checked');
+  });
+
+  it('check no filter glob input should show all file types by default', () => {
+    cy.get('.icon-button.fa-code')
+        .should('be.visible')
+        .first()
+        .click();
+
+    cy.get('.radio-button--search')
+        .should('be.visible')
+        .click();
+
+    // the contents of all the files should be visible
+    cy.get('#tab-authorship .files', { timeout: 90000 })
+        .should('be.visible');
+  });
+
+  it('check filter glob input should be in focus after click', () => {
+    cy.get('.icon-button.fa-code')
+        .should('be.visible')
+        .first()
+        .click();
+
+    cy.get('.radio-button--search')
+        .should('be.visible')
+        .click();
+
+    // click on input box
+    cy.get('#search')
+        .click()
+        .should('have.focus');
+  });
+
+  it('check filter glob input should have default empty value', () => {
+    cy.get('.icon-button.fa-code')
+        .should('be.visible')
+        .first()
+        .click();
+
+    cy.get('.radio-button--search')
+        .should('be.visible')
+        .click();
+
+    cy.get('#search')
+        .should('have.value', '');
+  });
+
+  it('check request to filter nothing by enter should show all file types', () => {
+    cy.get('.icon-button.fa-code')
+        .should('be.visible')
+        .first()
+        .click();
+
+    cy.get('.radio-button--search')
+        .should('be.visible')
+        .click();
+
+    // enter on empty input
+    cy.get('#search')
+        .type('{enter}');
+
+    cy.get('#tab-authorship .files', { timeout: 90000 })
+        .should('be.visible');
+  });
+
+  it('check request to filter nothing by clicking should show all file types', () => {
+    cy.get('.icon-button.fa-code')
+        .should('be.visible')
+        .first()
+        .click();
+
+    cy.get('.radio-button--search')
+        .should('be.visible')
+        .click();
+
+    // click 'Filter' on empty input
+    cy.get('#submit-button')
+        .click();
+
+    cy.get('#tab-authorship .files', { timeout: 90000 })
+        .should('be.visible');
+  });
+
+  it('check request to filter invalid glob by enter should not show any files', () => {
+    cy.get('.icon-button.fa-code')
+        .should('be.visible')
+        .first()
+        .click();
+
+    cy.get('.radio-button--search')
+        .should('be.visible')
+        .click();
+
+    cy.get('#tab-authorship .files', { timeout: 90000 })
+        .should('be.visible');
+
+    cy.get('#search')
+        .type('invalid glob');
+
+    cy.get('#submit-button')
+        .type('{enter}');
+
+    // no file should be shown
+    cy.get('#tab-authorship .files')
+        .should('not.be.visible');
+  });
+
+  it('check request to filter invalid glob by clicking should not show any files', () => {
```

Comment:
I feel that one test to check if click or enter works will do. After that, we can just test one of it. But this way is fine for me as well. Just that it felt a little redundant to do so.

Reply:
Hmm the reason I include test for click and enter both times for empty inputs and invalid inputs was that I think invalid inputs, we could check whether both click and enter work. 

For empty input, these tests were added because I remember the filter glob wasn't even clickable and enterable in the first place and this means that all files were present. Entering and clicking for empty inputs is to check whether the button and input work as expected. 

I lean a little towards including these tests, but feel free to let me know as well :)